driver/bpf/collector_probe.c

```diff
@@ -0,0 +1,290 @@
+ *        the parent and the child. This program is purely responsible for
+ *        stashing args for the child (which are a copy of the parent's)
+ *
+ *        These stashed args are used in subsequent process events where
+ *        these args are not available (e.g. fork).
+ */
+PROBE_SIGNATURE("sched/", sched_process_fork, sched_process_fork_args) {
+  struct sysdig_bpf_settings* settings;
+  enum ppm_event_type evt_type;
+  struct sys_stash_args args;
+  unsigned long* argsp;
+
+  settings = get_bpf_settings();
+  if (!settings) {
+    return 0;
+  }
+
+  if (!settings->capture_enabled) {
+    return 0;
+  }
+
+  // using the "private" version of this function so we can
+  // provide a pid.
+  argsp = __unstash_args(ctx->parent_pid);
+  if (!argsp) {
+    return 0;
+  }
+
+  memcpy(&args, argsp, sizeof(args));
+
+  __stash_args(ctx->child_pid, args.args);
+  return 0;
+}
+
+/**
+ * @brief program for handling sched_process_exit events. As the name suggests
+ *        they occur when a process exits. Minimal processing is performed here
+ *        instead, we defer to the appropriate filler.
+ */
+PROBE_SIGNATURE("sched/", sched_process_exit, sched_process_exit_args) {
+  struct sysdig_bpf_settings* settings;
+  enum ppm_event_type evt_type = PPME_PROCEXIT_1_E;
+  struct task_struct* task;
+  unsigned int flags;
+
+  task = (struct task_struct*)bpf_get_current_task();
+
+  flags = _READ(task->flags);
+  if (flags & PF_KTHREAD) {
+    // we only want to process userspace threads.
+    return 0;
+  }
+
+  settings = get_bpf_settings();
+  if (!settings) {
+    return 0;
+  }
+
+  if (!settings->capture_enabled) {
+    return 0;
+  }
+
+  call_filler(ctx, ctx, evt_type, settings, UF_NEVER_DROP);
+  return 0;
+}
+
+/**
+ * @brief Generic sys_enter_* program for any system call. It is responsible for
+ *        Verifying userspace settings and early processing of the syscall event.
+ *
+ *        The function will exit 0 (zero) regardless of outcome.
+ *
+ * @param id the syscall id
+ * @param ctx the context pointer as provided by the kernel
+ *
+ * @return 0 (regardless of outcomes)
+ */
+static __always_inline int enter_probe(long id, struct sys_enter_args* ctx) {
+  const struct syscall_evt_pair* sc_evt;
+  struct sysdig_bpf_settings* settings;
+  enum ppm_event_type evt_type = PPME_GENERIC_E;
+  int drop_flags = UF_ALWAYS_DROP;
+  struct sys_enter_args stack_ctx = {.id = id, {0}};
+
+  settings = get_bpf_settings();
+  if (!settings) {
+    return 0;
+  }
+
+  if (!settings->capture_enabled) {
+    // capture_enabled is usually false when we have loaded the probe
+    // but not started capturing from a userspace perspective.
+    return 0;
+  }
+
+  sc_evt = get_syscall_info(id);
+  if (!sc_evt) {
+    return 0;
+  }
+
+  if (sc_evt->flags & UF_USED) {
+    evt_type = sc_evt->enter_event_type;
+    drop_flags = sc_evt->flags;
+  } else {
+    // early indicator that this event is not needed/wanted, so just exit.
+    return 0;
+  }
```

Comment:
Do we need this block still? I thought we were going to only attach used syscalls and now I'm confused on why we are checking this.

Reply:
The only reason I kept it in was because we have userspace access to these settings and could still disable certain syscalls at runtime if required. Obviously we don't really support that now, but for the sake of this small check, it might be worth keeping in? what do you think?

Reply:
If we are not using it right now, maybe we can put it behind an `#ifdef`? That way we keep the code but don't execute it for the time being.

Reply:
I've decided to just remove it for now, we can always add it back in later if that requirement comes up, and without it the function is a little simpler